### ISSUE TITLE
[Claude Code] Build and deploy realtime WiFi/CSI ML human detection

### ISSUE THREAD

**Issue by Mokshith-GPT** [Student-Cybrarians/Human-Detection#2]
# Autonomous Claude Code build: production realtime WiFi/CSI human detection

Repository: https://github.com/Student-Cybrarians/Human-Detection
Current web target: https://human-detection.pages.dev/

## Mission

Use Claude Code and its available specialist agents/subagents to turn the existing JARVIS demo into a working, production-shaped realtime WiFi/CSI human-detection system.

Do not stop at architecture or mock UI. Inspect the repository, implement the system, run tests, deploy it, then verify the deployed application and report evidence.

## Non-negotiable truthfulness

- Do NOT fabricate hardware CSI, ML inference, distance, skeleton, latency, deployment, or test results.
- Simulator data must be explicitly marked as simulation.
- Do not display simulated distance as real distance.
- Real distance must be an ML/regression estimate with confidence/uncertainty and documented calibration requirements.
- Full human skeleton/pose must only be shown when a real trained model and appropriate CSI/hardware setup can produce it.
- If real hardware cannot be accessed in the execution environment, finish the complete hardware seam and a deterministic simulator/HIL path, deploy that path, and clearly label it.
- Do not claim the Cloudflare URL is hardware-live unless verified.

## Agent orchestration

Delegate explicitly where available:
1. backend-architect — realtime FastAPI/event architecture and contracts
2. python-pro / fastapi expert — implementation
3. ml-engineer / data-scientist — CSI preprocessing, PyTorch model, training/inference pipeline
4. ai-engineer — model serving/inference boundaries
5. frontend-developer — realtime JARVIS integration
6. ui-ux-designer — human body/skeleton visualization
7. test-automator / tdd-orchestrator — unit, integration, WebSocket, E2E
8. security-auditor — auth, CORS, WebSocket, secrets, dependency/security review
9. deployment-engineer / devops-troubleshooter — Docker, Cloudflare/backend deployment, logs
10. observability-engineer / performance-engineer — realtime metrics, latency, reconnects
11. code-reviewer — final review

Agents may work in parallel where safe, but keep a single authoritative event/schema contract.

## Required backend

Implement Python realtime services using PyTorch unless repository constraints justify TensorFlow.

Pipeline:

CSI source
-> acquisition adapter
-> amplitude/phase preprocessing
-> denoising/filtering/normalization
-> sliding temporal window
-> PyTorch inference
-> presence/activity/range outputs
-> optional pose/skeleton output
-> domain events
-> FastAPI/WebSocket
-> JARVIS frontend

Create clean interfaces for:
- CSIAdapter
- CSIFrame
- Preprocessor
- FeatureExtractor
- InferenceModel
- RangeEstimator
- PoseEstimator
- EventPublisher

Retain a simulator adapter for development and CI, but route it through the same preprocessing/inference/event pipeline rather than a separate fake-output path.

## ML requirements

Implement:
- reproducible synthetic CSI dataset generator for CI/development
- preprocessing for amplitude and phase
- temporal windowing
- PyTorch model for presence classification
- activity classification
- pluggable distance/range regression head
- checkpoint loading/versioning
- inference confidence
- model metadata/version in events
- training/evaluation scripts
- deterministic test fixtures

Use explicit output states such as unavailable/not-calibrated when a real range or pose model is not trained.

Distance must not be described as exact. Document that ranging requires calibrated CSI hardware geometry and labeled measurements.

Pose/skeleton:
- define a stable joint schema
- implement the estimator interface
- if a validated model is unavailable, return pose_available=false rather than inventing joints
- frontend should render the skeleton only from actual backend joint events

## Realtime contract

Preserve compatibility with existing events and extend them safely.

Example:

{
  "event": "human.detected",
  "event_id": "...",
  "sequence": 123,
  "timestamp": "...",
  "source": "simulator|hardware",
  "model": {"name": "jarvis-csi", "version": "..."},
  "person": {
    "count": 1,
    "confidence": 0.94,
    "distance_m": null,
    "distance_confidence": null,
    "activity": "walking",
    "activity_confidence": 0.91,
    "velocity_mps": null,
    "pose_available": false,
    "skeleton": []
  }
}

Also support signal.updated, human.lost, activity.changed, sensor health, model status, and connection status.

## Frontend

Update the deployed JARVIS UI to:
- clearly show LIVE / SIMULATION / DISCONNECTED state
- show model name/version and inference confidence
- show distance only when a real estimator provides it
- show distance confidence/uncertainty
- show activity and velocity when available
- show a human-body skeleton visualization driven exclusively by backend pose events
- preserve the existing cinematic JARVIS design
- never imply camera capture if this is CSI-only

## Deployment

Inspect the actual current Cloudflare setup and deployment configuration.

Required:
- production frontend at https://human-detection.pages.dev/
- production API origin reachable over HTTPS
- production WebSocket endpoint reachable over WSS
- correct CORS and WebSocket origin handling
- Dockerized backend
- environment-based configuration
- health/readiness endpoints
- CI build/test gates
- deployment logs/evidence

Cloudflare Pages cannot itself execute a persistent Python ML inference server. Put the Python/FastAPI ML backend on an appropriate persistent/container runtime and connect Pages to it through HTTPS/WSS. Do not pretend Pages is hosting the Python process.

If current Cloudflare Worker/reverse-proxy files exist, verify them rather than assuming they exist.

## Verification

Run, at minimum:
- backend unit tests
- ML preprocessing/model tests
- WebSocket tests
- simulator-to-inference integration test
- frontend tests/build
- E2E browser tests
- Docker build
- production health check
- deployed WebSocket smoke test
- realtime event smoke test

Prove the vertical slice:

CSI simulator/hardware adapter
-> preprocessing
-> PyTorch inference
-> human.detected
-> WebSocket
-> deployed JARVIS UI update

If hardware is physically available, run a real CSI capture and record the adapter/device used. If it is not available, explicitly report that hardware validation remains outstanding.

## Final acceptance report

Create/update docs with:
- architecture
- event schema
- ML model
- data/training requirements
- hardware setup
- calibration procedure
- deployment
- environment variables
- troubleshooting
- known limitations

At the end report:
1. files changed
2. agents used
3. tests run and exact results
4. deployed URLs
5. backend health result
6. WebSocket result
7. whether data is simulator or hardware
8. whether distance is actually ML-estimated
9. whether skeleton is actually ML-estimated
10. remaining blockers
11. commit/PR SHA

Do not mark the task complete merely because the frontend renders. The acceptance criterion is an evidenced realtime end-to-end path.

**Comment by Mokshith-GPT** [comment]
Claude Code execution handoff — please pick up this issue as the authoritative build specification.

Target production frontend: https://human-detection.pages.dev/
Repository default branch must be discovered from GitHub; do not assume `main`. Current GitHub metadata reports `master-branch`.

Deployment evidence supplied by project docs says a GitHub Actions workflow is intended to deploy the Pages frontend at the target URL and expects CLOUDFLARE_API_TOKEN/CLOUDFLARE_ACCOUNT_ID secrets. Verify the actual repository workflow/configuration before using that documentation. fileciteturn72file0L5-L14

Execute the implementation, tests, deployment, and verification—not just planning. At completion, post a comment here with exact commit/PR SHA, deployment URLs, CI results, backend health and WSS smoke-test evidence, and an explicit simulator-vs-real-hardware status.
